search-service/src/test/kotlin/com/egm/stellio/search/csr/service/ContextSourceCallerTests.kt

```diff
@@ -0,0 +1,190 @@
+                )
+        )
+
+        val response = ContextSourceCaller.getDistributedInformation(HttpHeaders.EMPTY, csr, path, emptyParams)
+        assertJsonPayloadsAreEqual(entityWithSysAttrs, serializeObject(response.getOrNull()!!))
+    }
+
+    @Test
+    fun `getDistributedInformation should return a MiscellaneousWarning if it receives no answer`() = runTest {
+        val csr = gimmeRawCSR()
+        val path = "/ngsi-ld/v1/entities/$apiaryId"
+
+        val response = ContextSourceCaller.getDistributedInformation(HttpHeaders.EMPTY, csr, path, emptyParams)
+
+        assertTrue(response.isLeft())
+        assertInstanceOf(MiscellaneousWarning::class.java, response.leftOrNull())
+    }
+
+    @Test
+    fun `getDistributedInformation should return a RevalidationFailedWarning when receiving a bad payload`() = runTest {
+        val csr = gimmeRawCSR()
+        val path = "/ngsi-ld/v1/entities/$apiaryId"
+        stubFor(
+            get(urlMatching(path))
+                .willReturn(
+                    ok()
+                        .withHeader(CONTENT_TYPE, APPLICATION_JSON_VALUE).withBody(entityWithBadPayload)
+                )
+        )
+
+        val response = ContextSourceCaller.getDistributedInformation(HttpHeaders.EMPTY, csr, path, emptyParams)
+
+        assertTrue(response.isLeft())
+        assertInstanceOf(RevalidationFailedWarning::class.java, response.leftOrNull())
+    }
+
+    @Test
+    fun `getDistributedInformation should return MiscellaneousPersistentWarning when receiving error 401`() = runTest {
+        val csr = gimmeRawCSR()
+        val path = "/ngsi-ld/v1/entities/$apiaryId"
+        stubFor(
+            get(urlMatching(path))
+                .willReturn(unauthorized())
+        )
+
+        val response = ContextSourceCaller.getDistributedInformation(HttpHeaders.EMPTY, csr, path, emptyParams)
+
+        assertTrue(response.isLeft())
+        assertInstanceOf(MiscellaneousPersistentWarning::class.java, response.leftOrNull())
+    }
+
+    @Test
+    fun `getDistributedInformation should return null when receiving an error 404`() = runTest {
+        val csr = gimmeRawCSR()
+        val path = "/ngsi-ld/v1/entities/$apiaryId"
+        stubFor(
+            get(urlMatching(path))
+                .willReturn(notFound())
+        )
+
+        val response = ContextSourceCaller.getDistributedInformation(HttpHeaders.EMPTY, csr, path, emptyParams)
+
+        assertTrue(response.isRight())
+        assertNull(response.getOrNull())
+    }
+
+    @Test
+    fun `getDistributedInformation should not ask context source for a GeoJson representation`() = runTest {
+        val csr = gimmeRawCSR()
+        val path = "/ngsi-ld/v1/entities/$apiaryId"
+        stubFor(
+            get(urlMatching(path))
+                .willReturn(ok())
+        )
+        val header = HttpHeaders()
+        header.accept = listOf(GEO_JSON_MEDIA_TYPE)
+        ContextSourceCaller.getDistributedInformation(
```

Comment:
Yes, because we can't merge different GEOJSON payload. moreover the GEOJSON is treated in the toFinalRepresentation which is after all the entity are merged.